pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -93,6 +98,19 @@ private MessageRouter getMessageRouter() {
 
         switch (messageRouteMode) {
             case CustomPartition:
+                String messageRouterClassName = conf.getMessageRouterClassName();
+                if (isNotBlank(messageRouterClassName)) {
+                    Class<?> clazz;
+                    try {
+                        clazz = Class.forName(messageRouterClassName);
+                        messageRouter = (MessageRouter) clazz.getDeclaredConstructor().newInstance();
+                        conf.setCustomMessageRouter(messageRouter);
+                    } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException |
+                            InstantiationException | IllegalAccessException | ClassCastException e) {
+                        log.error("[{}] Error occurred while instantiating the messageRouterClassName",
+                                messageRouterClassName, new PulsarClientException(e));
+                    }
```

Comment:
and yes, we should not eat the exception rather fail the producer creation.

Reply:
ok. then something like this should work I guess:
```java
log.error("[{}] Error occurred while instantiating the messageRouterFactoryClassName", messageRouterFactoryClassName);
throw new RuntimeException(e);
```

RuntimeException looks a little odd, but I can see that at many places, we have taken this approach.

@lhotari you also had suggestions on exceptions so please have a look here.